powershellgroup/powershellgroup.resource.ps1

```diff
@@ -55,45 +55,45 @@ Import-Module $DscModule
 
 if ($Operation -eq 'List')
 {
-    $DscResources= Get-DscResource
+
+    RefreshCache
     #TODO: following should be added to debug stream of every operation
     #$m = gmo PSDesiredStateConfiguration
     #$r += @{"DebugInfo"=@{"ModuleVersion"=$m.Version.ToString();"ModulePath"=$m.Path;"PSVersion"=$PSVersionTable.PSVersion.ToString();"PSPath"=$PSHome}}
     #$r[0] | ConvertTo-Json -Compress -Depth 3
-    foreach ($r in $DscResources)
+    foreach ($r in $script:ResourceCache.keys)
     {
-        if ($r.ImplementedAs -eq "Binary")
+        if ($script:ResourceCache[$r].ImplementedAs -eq "Binary")
```

Comment:
It would be better perf and will look slightly easier if `$script:ResourceCache[$r]` would be assigned to a var in the top of the loop and used in many places below.

Another idea: it is probably easier to iterate over cache values which are objects that we need/use below:
`foreach ($r in $script:ResourceCache.Values)`